File: ts/JobEligibilityPortal.ts
import * as readline from "readline";

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

const ask = (q: string): Promise<string> => new Promise(res => rl.question(q, res));

async function main() {
  console.log("Welcome to the Python Developer Job Portal!");
  console.log("-".repeat(45));

  const python = (await ask("Do you know Python? (yes/no): ")).trim().toLowerCase();
  const experience = parseInt((await ask("How many years of experience or projects do you have? ")).trim());
  const degree = (await ask("Do you have a CS degree or completed a Bootcamp? (yes/no): ")).trim().toLowerCase();

  if (python === "yes" && (experience >= 2 || degree === "yes")) {
    console.log("Congratulations! You have been accepted to the next interview stage.");
  } else {
    console.log("Sorry, your current qualifications do not match the job requirements.");
  }

  rl.close();
}

main();

File: rust/job_eligibility_portal.rs
use std::io::{self, Write};

fn read_line(prompt: &str) -> String {
    print!("{}", prompt);
    io::stdout().flush().unwrap();
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    input.trim().to_lowercase()
}

fn main() {
    println!("Welcome to the Python Developer Job Portal!");
    println!("{}", "-".repeat(45));

    let python = read_line("Do you know Python? (yes/no): ");
    let experience_str = read_line("How many years of experience or projects do you have? ");
    let experience: u32 = experience_str.trim().parse().expect("Invalid number");
    let degree = read_line("Do you have a CS degree or completed a Bootcamp? (yes/no): ");

    if python == "yes" && (experience >= 2 || degree == "yes") {
        println!("Congratulations! You have been accepted to the next interview stage.");
    } else {
        println!("Sorry, your current qualifications do not match the job requirements.");
    }
}
